src/components/UriViewer/LogViewer.ts

```diff
@@ -0,0 +1,224 @@
+        return { textContent: response.textContent, downloadUri: uri };
+      } catch (e) {
+        return null;
+      }
+    },
+    [signal]
+  );
+
+  useOnMount(() => {
+    fetchLogContent(logs[0].logUri).then((content) => {
+      if (_.isEmpty(content?.textContent)) {
+        setActiveTextContent(null);
+      } else {
+        setActiveTextContent(content?.textContent);
+      }
+      setActiveDownloadUri(content?.downloadUri);
+    });
+  });
+
+  useEffect(() => {
+    const fetch = async (logUri: string) => {
+      const res = await fetchLogContent(logUri);
+      if (_.isEmpty(res?.textContent)) {
+        setActiveTextContent(null); // Fetch failed. Display error.
+      } else {
+        setActiveTextContent(res?.textContent);
+      }
+      setActiveDownloadUri(res?.downloadUri);
+    };
+
+    // when switching tabs, reset content to undefined while we fetch the new content.
+    setActiveTextContent(undefined);
+    setActiveDownloadUri(undefined);
+
+    // tab switching set the currently active log (which triggers this effect). Fetch the content for the new log.
+    const uri = currentlyActiveLog?.logUri;
+    if (!_.isEmpty(uri)) {
+      fetch(uri);
+    }
+  }, [logs, currentlyActiveLog, fetchLogContent]);
+
+  const renderActiveTextContent = () => {
+    if (activeTextContent === undefined) {
+      return div([centeredSpinner()]);
+    }
+    if (activeTextContent === null) {
+      return div([
+        "Log file could not be loaded. If the workflow or task is still in progress, the log file likely hasn't been generated yet. Some logs may be unavailable if the workflow or task failed before they could be generated.",
+      ]);
+    }
+    return div([activeTextContent]);
+  };
+
+  const tabsArray: SimpleTabProps[] = logs.map((log) => {
+    return { key: log.logKey, title: log.logTitle, width: tabMaxWidth } as SimpleTabProps;
+  });
+
+  return h(
+    Modal,
+    {
+      onDismiss,
+      title: modalTitle,
+      titleChildren: h(
+        InfoBox,
+        {
+          style: { marginLeft: '1ch' },
+          tooltip: undefined,
+          size: undefined,
+          side: undefined,
+          iconOverride: undefined,
+        },
+        [
+          div({ style: { fontWeight: 'bold' } }, ['Execution:']),
+          div({ style: { marginLeft: '1rem', marginBottom: '1rem' } }, [
+            'Each workflow has a single execution log which comes from the engine running your workflow. Errors in this log might indicate a Terra systems issue, or a problem parsing your WDL.',
+          ]),
+          div({ style: { fontWeight: 'bold' } }, ['Task Standard Out/Error:']),
+          div({ style: { marginLeft: '1rem' } }, [
+            'Task logs are from user-defined commands in your WDL. You might see an error in these logs if there was a logic or syntax error in a command, or something went wrong while running it.',
```

Comment:
I left some suggestions for these descriptions in the doc but don't consider them blocking.